### PR TITLE
Signed messages format.

### PR DESCRIPTION
**Pull request by cgcardona**
A standard for interoperable generic signed messages based on the Bitcoin Script format and Ethereum format.

```txt
sign(sha256(length(prefix) + prefix + length(message) + message))
```

The prefix is simply the string '\x1AAvalanche Signed Message:\n', where 0x1A is the length of the prefix text and `length(message)` is an int of the message size.

### CODE REVIEW

File: build/references/cryptographic-primitives.md [hunk 1]
```diff
@@ -62,6 +62,43 @@ Message Signature: `0xb52aa0535c5c48268d843bd65395623d2462016325a86f09420c81f142
 
 Morty was never seen again.
 
+## Signed Messages
+
+A standard for interoperable generic signed messages based on the Bitcoin Script format and Ethereum format.
+
+```txt
+sign(sha256(length(prefix) . prefix . length(message) . message))
```
Comment (danlaine) [inline build/references/cryptographic-primitives.md:70]: Does `.` here denote the concatenation operator?

File: build/references/cryptographic-primitives.md [hunk 2]
```diff
@@ -62,6 +62,43 @@ Message Signature: `0xb52aa0535c5c48268d843bd65395623d2462016325a86f09420c81f142
...
++---------------+-----------+------------------------------+
+| messageLength : int       |                      4 bytes |
++---------------+-----------+------------------------------+
+| message       : []byte    |          size(message) bytes |
++---------------+-----------+------------------------------+
+                            |       26 + 4 + size(message) |
+                            +------------------------------+
+```
+
+### Example
```
Comment (danlaine) [inline build/references/cryptographic-primitives.md:89]: This is an example of the pre-image (the thing that will be hashed and then signed) right? For clarity, can we also include the result after hashing and signing it?

File: build/references/cryptographic-primitives.md [hunk 3]
```diff
@@ -62,6 +62,43 @@ Message Signature: `0xb52aa0535c5c48268d843bd65395623d2462016325a86f09420c81f142
...
+
+A standard for interoperable generic signed messages based on the Bitcoin Script format and Ethereum format.
+
+```txt
+sign(sha256(length(prefix) . prefix . length(message) . message))
+```
+
+The prefix is simply the string '\x1AAvalanche Signed Message:\n', where 0x1A is the length of the prefix text and `length(message)` is an int of the message size.
+
+### Gantt Signed Message Specification
```
Comment (danlaine) [inline build/references/cryptographic-primitives.md:75]: This is the format of the pre-image and not the actual signed message, right?

File: build/references/cryptographic-primitives.md (hunk 1 above)
Comment (cgcardona) [reply build/references/cryptographic-primitives.md:70]: Yes the `.` is the concatenation operator. I'll change it to `+` or clarity.

**Review by danlaine — APPROVED** [review]

**Pull request merged**